### PR TITLE
GCS does manual params validation

### LINKED ISSUE #4396: [gateway-gcs] GCS doesn't ask for projectID when not provided
## Expected Behavior
Error message should be gcs specific, there are no credentials provided with GCS.

## Current Behavior

Asks for users to set `Access and secret keys are mandatory to run Minio gateway server`

```
$ minio gateway gcs 
FATA[0000] Access and secret keys are mandatory to run Minio gateway server.  cause=Missing credentials source=[gateway-main.go:102:mustGetGatewayConfigFromEnv()]
```

## Possible Solution
Fix the error message.

## Steps to Reproduce (for bugs)
Just type `gcs` without arguments.
```
minio gateway gcs
``` 

## Context
GCS testing. 

## Your Environment
```
minio version
Version: 2017-05-23T02:27:58Z
Release-Tag: DEVELOPMENT.2017-05-23T02-27-58Z
Commit-ID: 37549f20b3b1d2d618f7944895c745b3c2092427
```

### PR DESCRIPTION
**Pull request by vadmeste**
## Description
To control the order of validation, gcs gateway checks project id
then access/secret keys.

```
$ minio gateway gcs
FATA[0000] Unable to initialize gateway layer            cause=GCS project ID expected source=[gateway-main.go:222:gatewayMain()]
```

## Motivation and Context
Fixed #4396 

## How Has This Been Tested?
go test + Manual

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/gateway-gcs.go
Comment (harshavardhana) [inline cmd/gateway-gcs.go]: `No credentials set` is correct instead of `No GCS credentials set`

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/gateway-gcs.go
Comment (vadmeste) [reply cmd/gateway-gcs.go]: oups yes.

Comment (vadmeste) [reply cmd/gateway-gcs.go]: > No credentials set is correct instead of No GCS credentials set

Done

**Comment by vadmeste** [comment]
Comments addressed and rebased.

**Comment by harshavardhana** [comment]
Can you rebase this PR @vadmeste ?

**Comment by vadmeste** [comment]
> Can you rebase this PR @vadmeste ?

Done

**Review by krishnasrinivas — APPROVED** [review]
Can you rebase @vadmeste

**Comment by vadmeste** [comment]
> Can you rebase @vadmeste

@krishnasrinivas: I rebased, but since this PR is quite old, we still need to see if we need it.

Currently this PR checks if gcs project id is provided and if access/secret keys are set in the environment. In contrary to S3 and Azure gateways which can work with credentials from config.json and not necessarily from the environment.

We need to check if we still need this.

**Comment by harshavardhana** [comment]
> n contrary to S3 and Azure gateways which can work with credentials from config.json and not necessarily from the environment.

Actually it is expected that for AWS S3 and Azure credentials are always provided through ENV regardless of them being saved in the config.json, GCS also has same requirements.

You can test and update us here if the feature-gcs branch is doing all the things which are handled here.

**Comment by vadmeste** [comment]
@harshavardhana, all gateways in feature-gcs branch don't require access and secret keys set in the environment.

**Comment by harshavardhana** [comment]
Then its a bug @vadmeste . All gateways require env to be set regardless of saving them in config file. Config file was introduced for configs like bucket notifications which we are going to support next. But setting envs is a must and it should fail otherwise.

**Comment by vadmeste** [comment]
> Then its a bug @vadmeste . All gateways require env to be set regardless of saving them in config file. Config file was introduced for configs like bucket notifications which we are going to support next. But setting envs is a must and it should fail otherwise.

Minio master requires access/secret keys from env when runnig s3 and azure, but feature-gcs branch doesn't. This PR also requires keys from env. So I guess things will be fine once feature-gcs got merged.

**Comment by vadmeste** [comment]
Overall, I think we can merge this PR now if no more comments.

**Pull request closed without merging**
